Source/WebKit/Platform/IPC/Connection.h

```diff
@@ -177,19 +177,48 @@ class Connection : public ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr<Connec
         virtual ~Client() { }
     };
 
+    struct Identifier;
+    struct Handle {
+        friend struct Identifier;
+        WTF_MAKE_NONCOPYABLE(Handle);
+        Handle() = default;
+        Handle(Handle&&) = default;
+        Handle& operator=(Handle&&) = default;
+
+#if USE(UNIX_DOMAIN_SOCKETS)
+        Handle(UnixFileDescriptor&& inHandle)
+            : handle(WTFMove(inHandle))
+        { }
+        explicit operator bool() const { return !!handle; }
+#elif OS(WINDOWS)
+        Handle(Win32Handle&& inHandle)
+            : handle(WTFMove(inHandle))
+        { }
+        explicit operator bool() const { return !!handle; }
+#elif OS(DARWIN)
+        Handle(MachSendRight&& sendRight)
+            : handle(WTFMove(sendRight))
+        { }
+        explicit operator bool() const { return MACH_PORT_VALID(handle.sendRight()); }
```

Comment:
`MachSendRight` has `operator bool` that only checks half of what `MACH_PORT_VALID` checks. (`MachSendRight::operator bool()` only checks against `MACH_PORT_NULL`)

Should `MachSendRight::operator bool()` be updated to use `MACH_PORT_VALID` and this code can be `explicit operator bool() const { return !!handle; }` in all cases?

Reply:
`MachSendRight` has explicit code for handing the 'dead name' value (the other half of MACH_PORT_VALID), including refcounting it. That said, I'm not sure if any consumer actually wants that behaviour.